static_frame/core/hloc.py

```diff
@@ -22,11 +24,37 @@ class HLoc(metaclass=HLocMeta):
             'key',
             )
 
-    def __init__(self, key: tp.Sequence[GetItemKeyType]):
+    def __init__(self, key: tp.Tuple[GetItemKeyType]) -> None:
         self.key = key
 
     def __iter__(self) -> tp.Iterator[GetItemKeyType]:
         return self.key.__iter__()
 
     def __len__(self) -> int:
         return self.key.__len__()
+
+    def __repr__(self) -> str:
+        def gen_nested_keys() -> tp.Iterator[str]:
+            for key in self.key:
+                if key.__class__ is not slice:
+                    yield str(key)
+                    continue
+
+                if key == NULL_SLICE:
+                    yield ':'
+                    continue
+
+                if key.start is None:  # type: ignore [union-attr]
+                    result = ':'
+                else:
+                    result = f'{key.start}:' # type: ignore [union-attr]
```

Comment:
More compact and only one type ignore!
```suggestion
                result = ':' if key.start is None else f'{key.start}' # type: ignore [union-attr]
```